### ISSUE TITLE
how can i imporve the model accuracy of FT4

### ISSUE THREAD

**Issue by 520jefferson** [NVIDIA/FasterTransformer#205]
model type: Transformer
batch size:4096 tokens
The task: chatbot task (Chinese to Chinese)
Training data: 100w samples
config1(enc36dec4)：enc 36 layer+dec4 layer+num_heads 16 + head size 64 + beam size 4
config(enc12dec12)：enc 12 layer+dec12 layer+num_heads 16 + head size 48 + beam size 4 (head size 48 is illegal)
config3(enc12dec12(standard)):enc 12 layer+dec12 layer+num_heads 12 + head size 64 + beam size 4

I compare the FT2.0 and FT4.0 accuracy(the proportion of same result between ft2.0 and ft4.0). the accuracy is realated about the depth of decoder（the difference between encoder is ver small) and the training steps.  The more deeper the decoder the less accuary between FT2.0 with FT4.0 . 
![image](https://user-images.githubusercontent.com/5691554/160313988-93d68f07-c801-4315-abd1-68ce1780fd50.png)

And i also compare the decoder results of FT2.0 and FT4.0, it's normal, the diff is very small.
![image](https://user-images.githubusercontent.com/5691554/160315120-edba3b75-cdaf-4639-b412-087a18705d1f.png)

**Comment by byshiue** [comment]
We have different kernels and different optimization in v4.0, that's why the results are different to v2.0. Especially for decoding model, when the model is deeper, the cumulative difference would be larger. 
You should compare them by a better metrics like accuracy of some task.
